xtask/src/ci.rs

```diff
@@ -289,15 +289,16 @@ impl CiTask {
         .map_err(Into::into)
     }
 
-    /// Lint ruma-common with clippy with the nightly version and wasm target.
+    /// Lint ruma-common and ruma-events with clippy with the nightly version and wasm target.
     ///
-    /// ruma-common is currently the only crate with wasm-specific code. If that changes, this
-    /// method should be updated.
+    /// ruma-common and ruma-events are currently the only crates with wasm-specific code. If that
+    /// changes, this method should be updated.
     fn clippy_wasm(&self) -> Result<()> {
         cmd!(
             "
             rustup run {NIGHTLY} cargo clippy --target wasm32-unknown-unknown
                 -p ruma-common --features api,js,rand
+                -p ruma-events --features unstable-msc1767,unstable-msc3488,unstable-msc3489,canonical-json
```

Comment:
I don't believe it will be too complicated to maintain a separate list of unstable features to check here. We would definitely forget to update it at some point.

My idea would be to split the `__ci` feature of the ruma crate to separate all the MSC features in a `__unstable-mscs` feature. Then have `__ci` enable it. And finally we can run here with `-p ruma --features api,js,rand,canonical-json,events,__unstable-mscs`

Reply:
What does `-p ruma` refer to in this context? Previously it was `ruma-common`, its not totally clear how to apply clippy to multiple crates. Additionally, I see events listed as one of the features, but I'm unsure if that is supposed to trigger the ruma-events crate or not.

Reply:
1. That's normal, that's exactly what we want. Since these types are marked as disallowed, they need to be replaced. I would add `Instant` as well, it's probably not used right now but we need to prevent its use as well. Since it's going to include a bunch of changes, maybe its better if we separate this change from the CI change in two separate PRs.
2. I was thinking of splitting the `__ci` feature from the ruma crate, not creating a new one in ruma-events.

Reply:
@zecakeh you're misunderstanding point 1, there are warnings for the *correct* types. I'm pretty certain that this js because `web_time` types are just re-exports of std types on non-wasm. We would need a separate clippy config that's only used for wasm to make use of the clippy lint here.

Reply:
Ah right, clippy is that smart. Yeah, I guess we need to use a different clippy config for WASM.

Reply:
So, I think one straight-forward solution for this since clippy doesn't support cfg-dependent config directly (AFAIK), would be to create `.clippy-wasm.toml` and copy it over to `.clippy.toml` before running xtask in CI (here: https://github.com/ruma/ruma/blob/7f562fe67efdc960fbc86a59207784efc1bbee1f/.github/workflows/ci.yml#L202)... Though it looks like we don't run `clippy-wasm` in CI at all? 😅

Reply:
Ah right, we don't. Well we need to add it to the CI then.

It's annoying that we have to rename the config file, it makes it harder to use locally. Who creates a binary with a config file where we can't provide a custom config file :stuck_out_tongue: ?

An alternative would be to use the [`CLIPPY_CONF_DIR`](https://doc.rust-lang.org/stable/clippy/configuration.html) env variable and place the alternate file in a path like `.wasm/clippy.toml`, but that's a bit ugly.

Reply:
Hm, maybe a bit more ugly but I'd say let's use the conf dir thing. Then it works in local xtask too.

Reply:
Thanks for all the good feedback! Ive updated the MR with `CLIPPY_CONF_DIR`. The last piece im missing is if I need to update the ci.yml with a specific job for clippy wasm? ex:

```
clippy-wasm:
    name: Clippy WASM
    needs: xtask
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repo
        uses: actions/checkout@v4

      - name: Install rust nightly toolchain
        uses: dtolnay/rust-toolchain@master
        with:
          toolchain: ${{ env.NIGHTLY }}

      - uses: Swatinem/rust-cache@v2

      - name: Set Clippy configuration directory
        run: export CLIPPY_CONF_DIR=$(pwd)/.wasm

      - name: Run Clippy for WASM
        run: rustup run nightly cargo clippy --target wasm32-unknown-unknown -p ruma --features api,js,rand,canonical-json,events,__unstable-mscs
```

Reply:
It should be possible to add it only with a new entry in the matrix of the nightly job.

Like adding this entry to the nightly matrix:

```yaml
          - name: Clippy WASM
            cmd: clippy-wasm
            targets: wasm32-unknown-unknown
            components: clippy
```

And change this step:

```yaml
      - name: Install rust nightly toolchain
        uses: dtolnay/rust-toolchain@master
        with:
          toolchain: ${{ env.NIGHTLY }}
          components: ${{ matrix.components }}
          targets: ${{ matrix.targets }}
```